lib/instrumentation/index.js

```diff
@@ -60,14 +61,40 @@ function Instrumentation (agent) {
       return this.bindingSpan || this.activeSpan
     }
   })
+
+  // NOTE: we need to track module names for patches
+  // in a separate array rather than using Object.keys()
+  // because the array is given to the hook(...) call.
+  this._patches = new NamedArray()
+
+  for (let mod of MODULES) {
+    this.addPatch(mod, require.resolve(`./modules/${mod}`))
+  }
+}
+
+Instrumentation.prototype.addPatch = function (name, handler) {
+  if (typeof handler !== 'function' && typeof handler !== 'string') {
+    this._agent.logger.error('Invalid patch handler type')
```

Comment:
To ease future debugging, I'd probably output the wrong type as well:

```suggestion
    this._agent.logger.error('Invalid patch handler type:', typeof handler)
```

Not sure how clever the V8 optimizer is, but you might also consider to only run the `typeof` once and store the result in a `const`

Reply:
fixed